File: ml-service/app/schemas/response.py
from pydantic import BaseModel
from typing import Dict, Optional, Any

class PredictResponse(BaseModel):
    model: str
    prediction: Any
    label: str
    probability: Optional[Dict[str, float]] = None
    inference_time_ms: float

class HealthResponse(BaseModel):
    status: str
    loaded_models: list[str]
    version: str

class ModelsResponse(BaseModel):
    models: list[Dict[str, Any]]

class ErrorResponse(BaseModel):
    error: str
    message: str
    available_models: Optional[list[str]] = None

File: be-service/internal/model/response.go
package model

type ClassifyResponse struct {
	Topic           string             `json:"topic"`
	Prediction      string             `json:"prediction"`
	Probability     map[string]float64 `json:"probability,omitempty"`
	InferenceTimeMs float64            `json:"inference_time_ms"`
}

type MLRequest struct {
	Features map[string]interface{} `json:"features"`
}

type MLResponse struct {
	Model           string             `json:"model"`
	Prediction      interface{}        `json:"prediction"` // Can be string or int
	Label           string             `json:"label"`
	Probability     map[string]float64 `json:"probability"`
	InferenceTimeMs float64            `json:"inference_time_ms"`
}

type ErrorResponse struct {
	Error   string `json:"error"`
	Message string `json:"message"`
}
